Code/RDGeneral/RDProps.h

```diff
@@ -124,26 +161,24 @@ class RDProps {
     from our list of \c computedProperties
   */
   //! \overload
+  void clearProp(int tag) const {
+    std::vector<int>::iterator it = std::find(compLst.begin(),
+                                              compLst.end(),
+                                              tag);
+    if (it != compLst.end())
+      compLst.erase(it);
+    dp_props.clearVal(tag);
+  };
+
   void clearProp(const std::string &key) const {
-    STR_VECT compLst;
-    if (getPropIfPresent(detail::computedPropName, compLst)) {
-      STR_VECT_I svi = std::find(compLst.begin(), compLst.end(), key);
-      if (svi != compLst.end()) {
-        compLst.erase(svi);
-        dp_props.setVal(detail::computedPropName, compLst);
-      }
-    }
-    dp_props.clearVal(key);
+    clearProp(Dict::tagmap.get(key));
   };
 
   //! clears all of our \c computed \c properties
   void clearComputedProps() const {
-    STR_VECT compLst;
-    if (getPropIfPresent(detail::computedPropName, compLst)) {
-      BOOST_FOREACH (const std::string &sv, compLst) { dp_props.clearVal(sv); }
+    for(std::vector<int>::iterator it=compLst.begin(); it!=compLst.end(); ++it)
```

Comment:
@greglandrum might have a better answer.  We do have a branch for exploring c++11 so we are (slowly) moving forwards.  We've already had to test for c++11 linkage on OS X because anaconda/boost can select either one depending on a lot of factors.